internal/tasks/repotester.go

```diff
@@ -41,38 +40,51 @@ func (r *RepoTester) TestRepositories() {
 
 	log.Info("Running RepoTester")
 
-	repositories, err := r.repoStore.ListIgnoreOrg()
-	if err != nil {
-		log.Errorf("error fetching repositories: %s", err)
-		return
-	}
-
-	for i := range repositories {
-		repository := repositories[i]
-
-		repoSpec, _ := repository.Spec.Data.GetGenericRepoSpec()
-		switch repoSpec.Type {
-		case "http":
-			log.Info("Detected HTTP repository type")
-			r.TypeSpecificRepoTester = &HttpRepoTester{}
-		case "git":
-			log.Info("Defaulting to Git repository type")
-			r.TypeSpecificRepoTester = &GitRepoTester{}
-		default:
-			log.Errorf("unsupported repository type: %s", repoSpec.Type)
+	limit := int32(ItemsPerPage)
+	continueToken := (*string)(nil)
+
+	for {
+		repositories, status := r.serviceHandler.ListRepositories(ctx, api.ListRepositoriesParams{
+			Limit:    &limit,
+			Continue: continueToken,
+		})
+		if status.Code != 200 {
+			log.Errorf("error fetching repositories: %s", status.Message)
+			return
 		}
 
-		accessErr := r.TypeSpecificRepoTester.TestAccess(&repository)
+		for i := range repositories.Items {
+			repository := repositories.Items[i]
+
+			repoSpec, _ := repository.Spec.GetGenericRepoSpec()
+			switch repoSpec.Type {
+			case "http":
```

Comment:
nit: could we make these consts :pray:

Reply:
Yes, for sure - especially when they're already defined :)